ext/date/php_date.c

```diff
@@ -2525,16 +2525,14 @@ PHPAPI bool php_date_initialize_from_ts_double(php_date_obj *dateobj, double ts)
 	zend_long sec;
 	int usec;
 
-	if (UNEXPECTED(isnan(sec_dval)
-		|| sec_dval >= (double)TIMELIB_LONG_MAX
-		|| sec_dval < (double)TIMELIB_LONG_MIN
-	)) {
-		zend_throw_error(
+	if (UNEXPECTED(isnan(sec_dval) || !PHP_DATE_DOUBLE_FITS_LONG(sec_dval))) {
+		zend_argument_error(
 			date_ce_date_range_error,
-			"Seconds must be a finite number between " TIMELIB_LONG_FMT " and " TIMELIB_LONG_FMT ", %g given",
+			1,
+			"must be a finite number between " TIMELIB_LONG_FMT " and " TIMELIB_LONG_FMT ".999999, %g given",
```

Comment:
Just an idea: we use the word `inclusive` in some error messages (one example: https://github.com/php/php-src/blob/8ef0e4cfa014a61fbbb2b4a8066a15c132d7fcac/ext/calendar/easter.c#L66). So we may use the word `exclusive` in case of the upper bound.